tools/CheckCommits.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Set \n as the only separator for lists since commit messages might
+# have spaces in them
+OLD_IFS=$IFS
+IFS=$'\n'
+
+# Get the last 300 commits and check those for bad messages. TravisCI doesn't
+# allow us to get develop (at least not easily) so we can't just compare find
+# the added commits.
+COMMIT_LINES=`git log -n 300 --oneline`
+
+# For all commit messages, check if they start with one of the key words
+for commit_msg in $COMMIT_LINES
+do
+    if grep "[0-9a-f]\{6\} [Ff][Ii][Xx][Uu][Pp]" <<< $commit_msg\
```

Comment:
Common comments about all the grep calls:
* Anchor regex at start.
* Once we start getting more commits we will have 6-character hash collisions and git will display extra characters to disambiguate, so use `\{6,40\}` (or even `*` since we know the format).